cmd/group-auto-updater/main.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/config/org"
+
+	v1 "github.com/openshift/api/user/v1"
+	userV1 "github.com/openshift/client-go/user/clientset/versioned/typed/user/v1"
+
+	"github.com/openshift/ci-tools/pkg/steps"
+	"github.com/openshift/ci-tools/pkg/util"
+)
+
+type options struct {
+	group           string
+	peribolosConfig string
+	org             string
+
+	dryRun bool
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.group, "group", "", "The group that will be updated in the cluster.")
+	fs.StringVar(&o.peribolosConfig, "peribolos-config", "", "Peribolos configuration file")
+	fs.StringVar(&o.org, "org", "", "Org from peribolos configuration")
+
+	fs.BoolVar(&o.dryRun, "dry-run", false, "Print the generated group without updating it")
+
+	fs.Parse(os.Args[1:])
+	return o
+}
+
+func validateOptions(o options) error {
+	if len(o.group) == 0 {
+		return fmt.Errorf("--group is not specified")
+	}
+	if len(o.peribolosConfig) == 0 {
+		return fmt.Errorf("--peribolos-config is not specified")
+	}
+	if len(o.org) == 0 {
+		return fmt.Errorf("--org is not specified")
+	}
+	return nil
+}
+
+func getUserV1Client() (*userV1.UserV1Client, error) {
+	clusterConfig, err := util.LoadClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not load cluster clusterConfig: %v", err)
+	}
+
+	userV1Client, err := userV1.NewForConfig(clusterConfig)
+	if err != nil {
+		return nil, fmt.Errorf("could not create user openshift client: %v", err)
+	}
+
+	return userV1Client, nil
+}
+
+func main() {
+	o := gatherOptions()
+	err := validateOptions(o)
+	if err != nil {
+		logrus.WithError(err).Fatal("invalid options")
+	}
+
+	logger := logrus.WithField("group", o.group)
+	dryLogger := steps.NewDryLogger(false)
+
+	b, err := ioutil.ReadFile(o.peribolosConfig)
+	if err != nil {
+		logger.WithError(err).Fatal("could not read peribolos configuration file")
+	}
+
+	var peribolosConfig org.FullConfig
+	if err := yaml.Unmarshal(b, &peribolosConfig); err != nil {
+		logger.WithError(err).Fatal("failed to unmarshal peribolos config")
+	}
+
+	userV1Client, err := getUserV1Client()
+	if err != nil {
+		logger.WithError(err).Fatal("could not get user client")
+	}
+
+	users := sets.NewString()
+	users.Insert(peribolosConfig.Orgs[o.org].Admins...)
+	users.Insert(peribolosConfig.Orgs[o.org].Members...)
+
+	g := &v1.Group{
+		ObjectMeta: metav1.ObjectMeta{Name: o.group},
+		Users:      users.List(),
+	}
+
+	group, err := userV1Client.Groups().Get(o.group, metav1.GetOptions{})
+	if err == nil {
+		group.Users = users.List()
+		dryLogger.AddObject(group.DeepCopyObject())
+
+		if _, err := userV1Client.Groups().Update(group); err != nil && !o.dryRun {
```

Comment:
But if we are dry-running, we should not actually do the `Update()` or `Create()` right? :)

Reply:
oops